### PR TITLE
filter invalid UTF-8 (including encoded surrogates) to make protobuf …

### PR DESCRIPTION
**Pull request by arnej27959**
…happy

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@geirst please review
@havardpe @vekterli FYI

if any of you know where we send "string" type fields through protobuf and it's possible that we can produce invalid UTF-8, we will need to add similar filtering there.

### CODE REVIEW

File: logd/src/logd/proto_converter.cpp [hunk 1]
```diff
@@ -59,7 +60,20 @@ ProtoConverter::log_message_to_proto(const LogMessage& message, ProtoLogMessage&
     proto.set_service(message.service());
     proto.set_component(message.component());
     proto.set_level(convert_level(message.level()));
-    proto.set_payload(message.payload());
+    const std::string &payload = message.payload();
+    vespalib::Utf8Reader reader(payload.c_str(), payload.size());
```
Comment (geirst) [inline logd/src/logd/proto_converter.cpp:64]: Consider moving the filtering to a helper function to make the code easier to read.

**Review by geirst — APPROVED** [review]
👍

File: logd/src/logd/proto_converter.cpp (hunk 1 above)
Comment (arnej27959) [reply logd/src/logd/proto_converter.cpp:64]: please take another look now

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**